### PR TITLE
[FLINK-6476] - Table environment register row data stream

### PR DESCRIPTION
**Pull request by rtudoran**
Added tests for registering row streams in java and scala

Thanks for contributing to Apache Flink. Before you open your pull request, please take the following check list into consideration.
If your changes take all of the items into account, feel free to open your pull request. For more information and/or questions please refer to the [How To Contribute guide](http://flink.apache.org/how-to-contribute.html).
In addition to going through the list, please provide a meaningful description of your changes.

- [x ] General
  - The pull request references the related JIRA issue ("[FLINK-XXX] Jira title text")
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message (including the JIRA id)

- [x ] Documentation
  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added

- [x ] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

### CODE REVIEW

**Comment by rtudoran** [comment]
@fhueske

Please have a look. I inserted tests in scala and java to check the registration of the stream.
For scala - it seemed to work directly if the implicit value is set (i did not know to test this before opening the JIRA).
For java - i have added the needed support in TableEnvironment.

I realized after the first open/closed PR that i forgot to test the case when row carried also names. Now this test is added and the corresponding support for it

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/TableEnvironment.scala [hunk 1]
```diff
@@ -677,6 +678,23 @@ abstract class TableEnvironment(val config: TableConfig) {
...
+        exprs.zipWithIndex flatMap {
+          case (UnresolvedFieldReference(name), idx) =>
+            Some((idx, name))
+          case (Alias(UnresolvedFieldReference(origName), name, _), _) =>
+            val idx = r.getFieldIndex(origName)
+            if (idx < 0) {
+              throw new TableException(s"$origName is not a field of type $r")
+            }
+            Some((idx, name))
+          case _ => throw new TableException(
```
Comment (fhueske) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/TableEnvironment.scala]: we need to add the `TimeAttribute` case here as well:
```
case _: TimeAttribute =>
  None
```

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/TableEnvironment.scala [hunk 2]
```diff
@@ -677,6 +678,23 @@ abstract class TableEnvironment(val config: TableConfig) {
           case _ => throw new TableException(
             "Field reference expression or alias on field expression expected.")
         }
+      case r: RowTypeInfo => {
+       // r.getFieldNames().map(name =>
```
Comment (fhueske) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/TableEnvironment.scala]: Please remove commented code

File: flink-libraries/flink-table/src/test/java/org/apache/flink/table/api/java/stream/sql/SqlITCase.java [hunk 3]
```diff
@@ -30,11 +30,125 @@
...
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.java.typeutils.RowTypeInfo;
 
 import java.util.ArrayList;
 import java.util.List;
 
 public class SqlITCase extends StreamingMultipleProgramsTestBase {
+	
+	@Test
+	public void testRowRegister() throws Exception {
```
Comment (fhueske) [inline flink-libraries/flink-table/src/test/java/org/apache/flink/table/api/java/stream/sql/SqlITCase.java]: This feature should not be tested with (so many) expensive integration tests.
Integration tests add a lot to the long build time and should be avoided if the feature can be tested with a more specific unit test. The `TableEnvironmentTest` has already many tests for `getFieldInfo`. We should add tests for `Row` there as well.

Comment (fhueske) [inline flink-libraries/flink-table/src/test/java/org/apache/flink/table/api/java/stream/sql/SqlITCase.java]: Adding one integration test for ingesting a `DataStream[Row]` to the scala `SqlITCase` would be good.

**Comment by rtudoran** [comment]
@fhueske thanks for the review. I will have make another pass and resubmit it

**Comment by rtudoran** [comment]
@fhueske 
I resubmitted it - please have a look. I hope it is fine now

**Comment by fhueske** [comment]
Thanks for the update @rtudoran.
+1 to merge

**Pull request closed without merging**